lib/braspag-rest/sale.rb

```diff
@@ -31,15 +31,16 @@ def save
     end
 
     def cancel(amount = nil)
-      response = BraspagRest::Request.void(request_id, payment.id, (amount || payment.amount))
+      response = BraspagRest::Request.void(request_id, payment.id, amount)
 
       if response.success?
-        self.payment.initialize_attributes(response.parsed_body)
+        voided_amount = amount ? payment.voided_amount.to_i + amount : payment.amount
+        self.payment.initialize_attributes(voided_amount: voided_amount)
```

Comment:
@bvicenzo Good catch. I've removed the return from the else branch to keep a single return point. See b4cd7cf.